### ISSUE TITLE
Gelbooru on Linux AppImage Not Loading Any Images

### ISSUE THREAD

**Issue by Velgen** [Bionus/imgbrd-grabber#2589]
**Bug description**

When trying to search for anything on Gelbooru it will show no images, but will allow you to change pages as if it had gotten results.

**Steps to reproduce**

1. Use the Linux AppImage
2. Set your source as Gelbooru
2. Search for any tag

**Expected behavior**

Showing results properly.

**Context**
Nothing in the log on startup seemed relevant so I decided to summarize it to what I get when I search which I think makes the problem quite clear.

```
[21:31:50.042][Info] Loading results...
[21:31:50.042][Info] [gelbooru.com][Html] Loading page https://gelbooru.com/index.php?page=post&s=list&tags=short_hair&pid=0
[21:31:50.043][Info] [gelbooru.com][Xml] Loading page https://gelbooru.com/index.php?page=dapi&s=post&q=index&limit=20&pid=0&tags=short_hair
[21:31:50.603][Info] [gelbooru.com][Html] Receiving page https://gelbooru.com/index.php?page=post&s=list&tags=short_hair&pid=0
[21:31:50.606][Info] [gelbooru.com][Html] Parsed page https://gelbooru.com/index.php?page=post&s=list&tags=short_hair&pid=0: 0 images (0), 61 tags (61), -1 total (1495494), 35607 pages (35607)
[21:31:50.695][Info] [gelbooru.com][Xml] Receiving page https://gelbooru.com/index.php?page=dapi&s=post&q=index&limit=20&pid=0&tags=short_hair
[21:31:50.696][Info] [gelbooru.com][Xml] Parsed page https://gelbooru.com/index.php?page=dapi&s=post&q=index&limit=20&pid=0&tags=short_hair: 0 images (0), 0 tags (0), 1568401 total (1568401), 78421 pages (78421)
```
It appears to be getting the total image and page count, but for some reason thinks there are no images. As stated prior I saw nothing related to Gelbooru on startup besides a failure to log in to Gelbooru which makes sense because I don't have an account setup.

**System information**

- OS: Linux
- Grabber version: 7.7.1

**Additional context**

Know there is a similar issue already, but I don't have any of the things those users have posted from their logs in mine so I decided to post mine as a separate issue. Site loads perfectly fine in any web browser on the same PC of course so it's definitely some kind of issue with Grabber of some kind.

**Comment by drakkillen** [comment]
have you logged in with your API key?

**Comment by Velgen** [comment]
Well I resolved the issue I thought after reading #2506 that since deleting the model data did not resolve the issue for those users that it was something more underlying, but it appears that at least on Linux it will not generate a new model data even if it's missing. You need to backup your config elsewhere, generate an entire new config folder/instance, then finally grab the model.js for Gelbooru from that config instance and replace your old one with it. Odd that Safebooru and Rule34 still worked for me considering they also use that model.

Makes me feel a little stupid, but at the same time it does seem a little dumb that you have to generate an entire new config to get it. You could either have it generate a new one if it finds it missing or have an option for the user to manually refresh the config/model for a site if they are experiencing issues. The latter of which from a usability standpoint is probably better than making users dig through the config files.

**Comment by Bionus** [comment]
Yes this is something that is being worked on, see 91aaf04b4a3cdaac676cd2355e37a6cd8497a2c9.

You can try on [Nightly](https://github.com/Bionus/imgbrd-grabber/releases/nightly) and see if it's solved.
The change is that the program does not copy the sources to the config directory anymore, and read them from the AppImage. You'll need to delete your existing config directory first though (or at least all "model.js" files in there), as they otherwise "override" the default ones.
